src/arcaflow_plugin_sdk/test_plugin.py

```diff
@@ -52,5 +52,25 @@ def cleanup():
         self.assertEqual("Hello world!\n", e.getvalue())
 
 
+@plugin.step(
+    "incorrect-return",
+    "Incorrect Return",
+    "A test that doesn't include the output ID.",
+    {"success": EmptyTestOutput},
+)
+def incorrect_return_step(
+    _: EmptyTestInput,
+):  # Skip return type, since we're purposefully not doing it right.
+    return EmptyTestOutput()
```

Comment:
Unless it's really onerous, I would recommend using the "correct" function signature and then using a linter directive to indicate that the bad return value is intentional.

Trying to skirt the issue by omitting the return value from the signature is likely to just create problems further up the food chain.  This test is supposed to test a failure to comply with the defined interface, so the interface should be "right", and it's the body which should be (explicitly) "wrong".

Reply:
From a Python perspective, the return value hint is non-functional -- adding it would not affect the operation of the test.

So, the purpose of the hint is to allow a tool like a linter to make sure that our interfaces match, that the call matches the definition.  These are supposed to match, regardless of the actual behavior of the function body.  

The fact that Dustin hit this probably stems from ignoring a lint complaint.  (We don't run a Python linter in the CI, do we?  Are we running Flake8?  Would it even catch this sort of thing?  Looking at `arcaflow-plugin-sdk-python/src/arcaflow_plugin_sdk/schema.py` in my IDE, it flags about 300 issues....)

In any case, dropping the return type hint because we know that we don't intend to honor the interface contract isn't the right approach.  In particular, even in the absence of the hint, it is possible for the linter to detect the return value type, at which point it could/would/should flag an error, presumably in the caller (which is probably not where we want to see it, since the "bug" is in the function body).  In order to prevent that, we need to tell the linter that we know that the code is erroneous.

We have two options:  we can either use the type hint corresponding to the actual function return type (and then, presumably, put a directive in the caller to suppress the warning there), or we can use the type hint which corresponds to the type which the caller properly expects and put a directive at the return point where we return the bad value.

The latter approach seems like the appropriate one to me:  since it is the function which is misbehaving, we shouldn't have to flag the caller; and, putting the directive in the function is code-as-documentation.

(If you're concerned about how to "spell" the lint directive, just use whatever works for you and the CI -- anyone else who hits it will see the directive (code as documentation), and they can add their own linter's directive to yours.)

Reply:
Okay. I found it here. https://gist.github.com/pylover/7870c235867cf22817ac5b096defb768